### ISSUE TITLE
CPU-mode: Wrong conditional structure for generated SCA column code

### ISSUE THREAD

**Issue by mlange05** [claw-project/claw-compiler#355]
For CPU mode, the generated loops along the horizontal/vector dimension in SCA code don't seem to deal correctly with conditionals that include column arrays.

**Original code**
```fortran
  !$claw define dimension proma(1:nproma) &                                                                                                                                                                                                                                   
  !$claw parallelize                                                                                                                                                                                                                                                          
  DO k = 1, nz
    IF (t(k) > 0.) THEN
      q(k) = q(k) / t(k)
    END IF
  END DO
```

**Transformation code**
```fortran
  DO k = 1 , nz , 1
   IF ( t ( proma , k ) > 0. ) THEN
    DO proma = 1 , nproma , 1
     q ( proma , k ) = q ( proma , k ) / t ( proma , k )
    END DO
   END IF
  END DO

```

**Expected code**
I'm not entirely sure how to deal with these conditionals gracefully, but using the following test code I get either a segfault due to an uninitialised `proma` in the conditional or a floating-point exception due to a divide-by-zero on the generated inner `nproma` loop. The test code is adapted from `main.f90` in the `abstraction1` test:
```
PROGRAM test_column_conditional
  USE mo_column, ONLY: compute_column
  REAL, DIMENSION(5,10) :: q, t  ! Fields as declared in the whole model                                                                                                                                                                                                        
  INTEGER :: nproma, nz           ! Size of array fields                                                                                                                                                                                                                        
  INTEGER :: p                    ! Loop index                                                                                                                                                                                                                                  

  nproma = 5
  nz = 10

  DO p = 1, nproma
    q(p, 1:nz) = 6.0
    t(p, 1:6) = 2.0
    t(p, 6:nz) = 0.0
  END DO

  !$claw parallelize forward                                                                                                                                                                                                                                                    
  DO p = 1, nproma
    CALL compute_column(nz, q(p,:), t(p,:))
  END DO

  PRINT*, (q(1, i), i=1,10)
END PROGRAM test_column_conditional
```

When run without Claw, I get (as expected):
```
3.00000000       3.00000000       3.00000000       3.00000000       3.00000000       6.00000000       6.00000000       6.00000000       6.00000000       6.00000000 
```

**Execution of `clawfc`**
```bash
clawfc --target=cpu --directive=none -o mo_column.claw.cpu.f90 mo_column.f90)
clawfc --target=cpu --directive=none -o main.claw.cpu.f90 main.f90)
```

**Comment by clementval** [comment]
@mlange05 You pointed out a good one! I have to re-design how we decide to place the `DO` statement in the CPU backend. Anyway, it was something I was thinking for a while and now that we pointed a bug, I'll start to do it.

**Comment by clementval** [comment]
Fixed by #361
